app/models/service.rb

```diff
@@ -105,11 +105,17 @@ def power_states
   end
 
   def custom_actions
-    service_template&.custom_actions(self)
+    if service_template.nil?
+      buttons = CustomButton.where(:applies_to_class => "Service")
+      button_groups = CustomButtonSet.all.select { |f| f.set_data[:applies_to_class] == "Service" }
+      {:buttons => buttons, :button_groups => button_groups}
```

Comment:
Looking at [app/models/mixins/custom_actions_mixin.rb#L11-L20](https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/custom_actions_mixin.rb#L11-L20) there looks like there is more to this.

Could the `Service` class include the mixin and then call it when `service_template` is not available?  I think it would avoid rewriting this logic.

Reply:
Then since I'm using the mixin logic should the custom_action_button query just be generic_custom_buttons from the mixin here: https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/custom_actions_mixin.rb#L52 ?